api/v1alpha1/edgedevice_types.go

```diff
@@ -34,9 +34,34 @@ type EdgeDeviceSpec struct {
 	// RequestTime is the time of device registration request
 	RequestTime *metav1.Time `json:"requestTime,omitempty"`
 
-	Heartbeat *HeartbeatConfiguration `json:"heartbeat,omitempty"`
-	Storage   *Storage                `json:"storage,omitempty"`
-	Metrics   *MetricsConfiguration   `json:"metrics,omitempty"`
+	Heartbeat     *HeartbeatConfiguration         `json:"heartbeat,omitempty"`
+	Storage       *Storage                        `json:"storage,omitempty"`
+	Metrics       *MetricsConfiguration           `json:"metrics,omitempty"`
+	LogCollection map[string]*LogCollectionConfig `json:"log_collection,omitempty"`
+}
+
+type LogCollectionConfig struct {
+
+	// Kind is the type of log collection to be used
+	// +kubebuilder:validation:Enum=syslog
+	Kind string `json:"kind,omitempty"`
+
+	// +kubebuilder:default=12
+	// +kubebuilder:validation:Minimum=1
+	BufferSize int32 `json:"BufferSize,omitempty"`
+
+	// SyslogConfig is the type to define Syslog Kind transport
+	SyslogConfig *LogSyslogConfig `json:"syslogConfig,omitempty"`
+}
+
+type LogSyslogConfig struct {
+
+	// Address with port to connect to syslog server.
+	Address string `json:"address,omitempty"`
```

Comment:
WDYT of replacing it with a configmap (and for credentials when needed a secret), so when that configuration changes, the admin have to patch a single CR instead of all of the devices?